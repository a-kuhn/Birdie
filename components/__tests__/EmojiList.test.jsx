import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import EmojiList from "../EmojiList";

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe("EmojiList", () => {
  const mockOnSelect = jest.fn();
  const mockOnCloseModal = jest.fn();

  it("renders correctly with all emojis", () => {
    const { getAllByTestId } = render(
      <EmojiList onSelect={mockOnSelect} onCloseModal={mockOnCloseModal} />
    );

    const images = getAllByTestId(/emoji-image-/);
    expect(images).toHaveLength(6);
  });

  it("calls onSelect and onCloseModal when an emoji is pressed", () => {
    const { getAllByRole } = render(
      <EmojiList onSelect={mockOnSelect} onCloseModal={mockOnCloseModal} />
    );

    const pressable = getAllByRole("button")[0];
    fireEvent.press(pressable);

    expect(mockOnSelect).toHaveBeenCalledWith(expect.any(Number));
    expect(mockOnCloseModal).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const { toJSON } = render(
      <EmojiList onSelect={mockOnSelect} onCloseModal={mockOnCloseModal} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
