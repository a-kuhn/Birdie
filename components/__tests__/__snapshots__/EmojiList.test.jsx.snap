// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiList matches snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
  data={
    [
      1,
      1,
      1,
      1,
      1,
      1,
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-0"
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-1"
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-2"
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-3"
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-4"
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          accessibilityRole="image"
          source={1}
          style={
            {
              "height": 100,
              "marginRight": 20,
              "width": 100,
            }
          }
          testID="emoji-image-5"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
